### PR TITLE
Fix default cloud region regex to not extract regions with slashes

### PR DESCRIPTION
**Pull request by glrf**
The updated regex will not extract wrong regions from subdirectory in cloud directory.

I.E `cloud/cloudscale/foo/bar` will not get any region instead of region `foo/bar`.

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [ ] Update tests.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog

### CODE REVIEW

File: src/commodore/index.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const defaultExtraConfig = {
   distributionRegex:
     /^distribution\/(?<distribution>[^\/]+)(?:\/cloud\/(?<cloud>.+)\.ya?ml|(?:\/.+)?\.ya?ml)$/,
   cloudRegionRegex:
-    /^cloud\/(?<cloud>[^\/]+)(?:\/(?<region>.+)\.ya?ml|\.ya?ml)$/,
+    /^cloud\/(?<cloud>[^\/]+)(?:\/(?<region>[^\/]+)\.ya?ml|.*\.ya?ml)$/,
```
Comment (simu) [inline src/commodore/index.ts:41]: I'd prefer to not match on `cloud/<cloudname>/.yaml`, but it probably doesn't make a large difference.

```suggestion
    /^cloud\/(?<cloud>[^\/]+)(?:\/(?<region>[^\/]+)\.ya?ml|.+\.ya?ml)$/,
```

**Review by simu — COMMENTED** [review]
This PR should also update https://github.com/projectsyn/commodore-renovate/blob/master/src/commodore/readme.md#extra-configuration-file

File: src/commodore/index.ts (hunk 1 above)
Comment (glrf) [reply src/commodore/index.ts]: That doesn't match `cloud/foo.yaml`. The following should work though `/^cloud\/(?<cloud>[^\/]+)(?:\/(?<region>[^\/]+)\.ya?ml|[^\/]*\.ya?ml)$/`

File: src/commodore/index.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const defaultExtraConfig = {
   distributionRegex:
     /^distribution\/(?<distribution>[^\/]+)(?:\/cloud\/(?<cloud>.+)\.ya?ml|(?:\/.+)?\.ya?ml)$/,
   cloudRegionRegex:
-    /^cloud\/(?<cloud>[^\/]+)(?:\/(?<region>.+)\.ya?ml|\.ya?ml)$/,
+    /^cloud\/(?<cloud>[^\/]+)(?:\/(?<region>[^\/]+)\.ya?ml|[^\/]*\.ya?ml)$/,
```
Comment (simu) [inline src/commodore/index.ts:41]: I think the actual new regex should be

```suggestion
    /^cloud\/(?<cloud>[^\/]+)(?:\/(?:(?<region>[^\/]+)|.+)\.ya?ml|\.ya?ml)$/,
```

the last `\.ya?ml` was originally only supposed to match files like `cloud/<cloudname>.yaml`, but with the restricted region regex we need to add an additional option to match everything which has prefix `cloud/<cloudname>/` but doesn't match `cloud/<cloudname>/<name>.ya?ml` in the second top-level group.

**Review by simu — COMMENTED** [review]
Probably also worth it to add the new cases in `src/commodore/util.spec.ts`

**Comment by glrf** [comment]
After conceding that I can't just quickly fix a regex I added some test cases for some edge cases I could come up with

File: src/commodore/util.spec.ts [hunk 3]
```diff
@@ -32,7 +32,18 @@ describe('src/commodore/util', () => {
...
+      ${'cloud/cloud2/region2.yaml'}         | ${{ distribution: null, cloud: 'cloud2', region: 'region2' }}
       ${'cloud/cloud/params.yml'}            | ${{ distribution: null, cloud: 'cloud', region: null }}
+      ${'cloud/cloud2/params.yaml'}          | ${{ distribution: null, cloud: 'cloud2', region: null }}
+      ${'cloud/cloud/params/param.yml'}      | ${{ distribution: null, cloud: 'cloud', region: null }}
+      ${'cloud/cloud/foo/bar.yml'}           | ${{ distribution: null, cloud: 'cloud', region: null }}
+      ${'cloud/cloud2/foo/bar/buzz.yaml'}    | ${{ distribution: null, cloud: 'cloud2', region: null }}
+      ${'cloud/cloud2/foo/.yaml'}            | ${{ distribution: null, cloud: 'cloud2', region: null }}
+      ${'cloud/cloud//.yml'}                 | ${{ distribution: null, cloud: 'cloud', region: null }}
+      ${'cloud/cloud/.yml'}                  | ${{ distribution: null, cloud: null, region: null }}
+      ${'cloud/cloudnt/.yaml'}               | ${{ distribution: null, cloud: null, region: null }}
```
Comment (glrf) [inline src/commodore/util.spec.ts:44]: I'm not quite sure what the correct behavior for these cases would be. I mean they are quite esoteric, so I don't know it it matters, but it feels weird to match `cloud/<cloud>/foo/.yml` and not match `cloud/<cloud>/.yaml`.

Comment (simu) [reply src/commodore/util.spec.ts:44]: I agree that these cases are not very consistent, but I don't think it matters too much, since users are free to adjust the regex if they need different behaviors. Side-note (not relevant to thit PR): I'd strongly discourage users from actually using `.yml` or `.yaml` as a reclass class (I'm not sure you'd even be able to include such a class...)

**Review by simu — APPROVED** [review]
